Purpose: Prevent Required-Stop: $all from modifying start links
Fixes:   #485307
Status:  Work in progress.
---
--- a/insserv.c
+++ b/insserv.c
@@ -291,8 +291,13 @@
 	    break;
 	case '$':
 	    if (strcasecmp(token, "$all") == 0) {
-		serv->attr.flags |= SERV_ALL;
-		break;
+		if (bit & REQ_KILL) {
+		    /* $all has no effect on stop sort order */
+		    break;
+		} else {
+		    serv->attr.flags |= SERV_ALL;
+		    break;
+		}
 	    }
 	    /* Expand the `$' token recursively down */
 	    list_for_each(ptr, sysfaci_start) {
